SolveNQueens

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity >=0.4.16 <0.9.0;
+pragma experimental ABIEncoderV2;
+
+contract SolveNQueens {
+
+    TemplateTestContract testContract;  // Create variable for the testContract which needs to be solved.
+    address payable owner;      
+    bool[][]board;              // Chessboard
+    uint8 N;                    // Number of queens  
+
+    constructor(address testAddress) payable {              
+        testContract = TemplateTestContract(testAddress);   // Initialise the testContract variable.
+        owner = msg.sender;                                 // Initialise the owner of the contract to be the creator of the contract.
+        N = 4;
+    }
+
+     // Function to solve the testContract.
```

Comment:
Does the function solve the testContract, or does it solve the task specified in the test contract?